docs/adr/0005-remote-settings-client.md

```diff
@@ -0,0 +1,112 @@
+* The client itself does not persist a local cache of remote resources, but instead delegates this responsibility to the consuming application (in this case, nimbus itself, which does persist them via the [rkv library](https://github.com/mozilla/rkv))
+* It does not use async Rust, but instead everything is blocking and run on threads exclusively created by the app itself.
+* It uses viaduct for all networking, meaning that wherever possible, Necko is used for networking, which is important for Android.
+
+## Considered Options
+### Option 1: Writing a new library
+The requirements of this client are such that writing new libraries in Kotlin and Swift is currently a realistic option.
+However, we are rejecting this option because we don’t want to duplicate the effort required to write and maintain two libraries - inevitably, the features and capabilities will diverge.
+Future requirements such as supporting content signature verification would lead to significant duplication.
+
+Writing a new library from scratch in Rust and exposing it via UniFFI so it can be used by both platforms is also a possibility.
+However, we are rejecting this option because existing Rust libraries already exist, so we would be better served by modifying or forking one of the existing libraries.
+
+### Option 2: Use the existing remote settings client
+Modifying or forking the existing client is an attractive option.
+It would require significant changes - the async capabilities would probably need to be removed (using a Rust event loop in our mobile browsers is something we are trying to avoid until we better understand the implications given these browsers already have an event loop and their own threading model).
+
+The persistence model used by this library is something that is not a requirement for the new library, which isn’t itself a problem, but it probably would preclude being able to use this library by Nimbus - so the end result is that we would effectively have two remote-settings clients written in Rust and used by our browsers.
+The networking code would need to be replaced with viaduct, which would not be a significant problem assuming we have dropped all async code.
+
+Some API changes would probably be required to make it suitable for use by UniFFI would also be necessary, but these would be tractable.
+
+We would need to update nimbus to use this client, and would need to move the client into the application-services repository, otherwise we would have issues with dependency management (ie, ensuring the upstream version changes continued to work for us, and managing the circular dependency which exists due to this library needing to use viaduct) and would complicate our build process (the library would need to be in our “megazord”, so pulling it remotely would complicate CI and our release process).
+These are the exact reasons why Nimbus itself is in the application-services repo.
```

Comment:
Was there a document covering the proposal to move to application-service, linking to that here will provide good context!

Reply:
3c356141e9177b963fbaf31aa07e9196b343cc27